influxql/iterator.gen.go

```diff
@@ -494,6 +494,103 @@ func (itr *floatParallelIterator) monitor() {
 	}
 }
 
+type lazyFloatIterator struct {
+	itr FloatIterator
+	fn  func() (Iterator, error)
+}
+
+// init instantiates the underlying iterator.
+func (itr *lazyFloatIterator) init() error {
+	if itr.itr != nil {
+		return nil
+	}
+
+	i, err := itr.fn()
+	if err != nil {
+		return err
+	} else if i == nil {
+		return nil
+	}
+
+	if it, ok := i.(FloatIterator); ok {
+		itr.itr = it
+	} else {
+		return fmt.Errorf("invalid lazy float iterator type: %T", i)
+	}
+
+	return nil
+}
+
+func (itr *lazyFloatIterator) Close() error {
+	if itr.itr == nil {
+		return nil
+	}
+	return itr.itr.Close()
+}
+
+func (itr *lazyFloatIterator) Next() (*FloatPoint, error) {
+	if err := itr.init(); err != nil {
+		return nil, err
+	} else if itr.itr == nil {
+		return nil, nil
+	}
+	return itr.itr.Next()
+}
+
+func (itr *lazyFloatIterator) Stats() IteratorStats {
```

Comment:
Could this break the series limit? It seems to me like this would not work anymore.